examples/unity/js/UnityTransformer.ts

```diff
@@ -0,0 +1,155 @@
+                    frameworkUrl: window.location.href + "/demoBuild.framework.js",
+                    codeUrl: window.location.href + "/demoBuild.wasm",
+                };
+                //@ts-ignore
+                createUnityInstance(unityCanvas, config, (progress: number) => {
+                }).then((unityInstance: any) => {
+                    this.unityGame_ = unityInstance;
+                    resolve()
+                }).catch((message: any) => {
+                    reject(message);
+                });
+            };
+            unityCanvas.appendChild(unityScript)
+        })
+    }
+
+    start() {
+
+    }
+
+    async transform(frame: VideoFrame, controller: TransformStreamDefaultController) {
+        if (this.videoFrameCanvas_.width != frame.displayWidth || this.videoFrameCanvas_.height != frame.displayHeight) {
+            this.videoFrameCanvas_.width = frame.displayWidth
+            this.videoFrameCanvas_.height = frame.displayHeight
+            this.unityimageData_ = new ImageData(frame.displayWidth, frame.displayHeight)
+        }
+
+        if (this.resultCanvas_.width != frame.displayWidth || this.resultCanvas_.height != frame.displayHeight) {
+            this.resultCanvas_.width = frame.displayWidth
+            this.resultCanvas_.height = frame.displayHeight
+        }
+
+        if (this.unityGame_) {
+            const timestamp: number = frame.timestamp ? frame.timestamp : Date.now();
+            //@ts-ignore
+            var VonageUnity: any = globalThis.VonageUnity;
+            if (typeof VonageUnity === "object") {
+                createImageBitmap(frame).then(image => {
+                    frame.close()
+                    this.processFrame(image)
+                    //@ts-ignore
+                    controller.enqueue(new VideoFrame(this.resultCanvas_, { timestamp, alpha: 'discard' }));
+                }).catch(e => {
+                    console.error("createImageBitmap", e);
+                    controller.enqueue(frame)
+                })
+            }
+        } else {
+            controller.enqueue(frame)
+        }
+    }
+
+    processFrame(image: ImageBitmap) {
+
+        this.videoFrameCtx_!.drawImage(
+            image,
+            0,
+            0,
+            image.width,
+            image.height,
+            0,
+            0,
+            this.videoFrameCanvas_.width,
+            this.videoFrameCanvas_.height
+        )
+        let imageData = this.videoFrameCtx_!.getImageData(
+            0,
+            0,
+            this.videoFrameCanvas_.width,
+            this.videoFrameCanvas_.height
+        )
+
+        if (this.unityGame_) {
+            //@ts-ignore
+            var VonageUnity: any = globalThis.VonageUnity;
+            if (typeof VonageUnity === "object") {
+                for (let i = 0; i < imageData.data.length; i += 4) {
+                    this.unityGame_.Module.HEAPF32[(VonageUnity.input.array >> 2) + i] = imageData.data[i] / 255
```

Comment:
Cannot we use `HEAP32` or even`HEAPU32` so we avoid floating point operations when writing and reading data to and from unity?

Reply:
https://docs.unity3d.com/ScriptReference/Color.html is taking float.... we can do conversions but its a demo.